experimental/Schemes/Types.jl

```diff
@@ -985,3 +987,138 @@ mutable struct VarietyFunctionFieldElem{FracType<:AbstractAlgebra.Generic.Frac,
+        # extract the candidates for the localization element
+        candidates = dual_facet.pm_cone.HILBERT_BASIS_GENERATORS[2]
+        v = candidates[1,:]
+        for j in 1:nrows(candidates)
+          v = candidates[j,:]
+          if contains(CXdual, v) && contains(CYdual, -v)
+            break
+          elseif contains(CXdual, -v) && contains(CYdual, v)
+            v = -v
+            break
+          end
+        end
+        (contains(CXdual, v) && contains(CYdual, -v)) || error("no element found for localization")
+
+        # We have found it and it is v. 
+        # Express v (resp. -v) in the hilbert bases of the two patches 
+        # and localize at the corresponding elements. 
+        vmat = matrix(ZZ.(collect(v)))
+        AX = transpose(hilbert_basis(X))
+        Id = identity_matrix(ZZ, ncols(AX))
+        sol = solve_mixed(AX, vmat, Id, zero(vmat))
+
+        fX = prod([x^k for (x, k) in zip(gens(ambient_ring(X)), sol)])
+        U = PrincipalOpenSubset(X, OO(X)(fX))
+        
+        AY = transpose(hilbert_basis(Y))
+        Id = identity_matrix(ZZ, ncols(AY))
+        sol = solve_mixed(AY, -vmat, Id, zero(vmat))
+
+        fY = prod([x^(k>0 ? 1 : 0) for (x, k) in zip(gens(ambient_ring(Y)), sol)])
+        V = PrincipalOpenSubset(Y, OO(Y)(fY))
+        
+        # set up the glueing isomorphisms 
+        # We assume that the v element must be part of the Hilbert basis 
+        # already. 
+        l = 0
+        for j in 1:ncols(AX)
+          if vmat == AX[:, j] 
+            l = j
+            break
+          end
+        end
+        Idext = identity_matrix(ZZ, ncols(AX))
+        Idext[l,l] = 0
+        img_gens = [solve_mixed(AX, AY[:, k], Idext, zero(MatrixSpace(ZZ, ncols(Idext), 1))) for k in 1:ncols(AY)]
+        fres = hom(OO(V), OO(U), [prod([(k >= 0 ? x^k : inv(x)^(-k)) for (x, k) in zip(gens(OO(U)), w)]) for w in img_gens])
+        
+        l = 0
+        for j in 1:ncols(AY)
+          if -vmat == AY[:, j] 
+            l = j
+            break
+          end
+        end
+        Idext = identity_matrix(ZZ, ncols(AY))
+        Idext[l,l] = 0
+        img_gens = [solve_mixed(AY, AX[:, k], Idext, zero(MatrixSpace(ZZ, ncols(Idext), 1))) for k in 1:ncols(AX)]
+        gres = hom(OO(U), OO(V), [prod([(k >= 0 ? x^k : inv(x)^(-k)) for (x, k) in zip(gens(OO(V)), w)]) for w in img_gens])
+
+        set_attribute!(gres, :inverse, fres)
+        set_attribute!(fres, :inverse, gres)
+        f = SpecMor(U, V, fres)
+        g = SpecMor(V, U, gres)
+        set_attribute!(g, :inverse, f)
+        set_attribute!(f, :inverse, g)
+
+        G = SimpleGlueing(X, Y, f, g)
+        add_glueing!(cov, G)
+      end
+    end
+
+    # TODO: Investigate whether or not we can make use of the Hasse Diagram 
+    # to take shortcuts in filling up the remaining glueings. 
+    # We conjecture, that the composition of the already existing glueings 
+    # is already maximal and so a lot of lengthy computations can be skipped.
```

Comment:
@lkastner If I remember correctly, you were looking into how one could make use of the Hasse diagram for chasing and extracting glueings, right? Let us meet again one of these days and try to work this stuff out.